File: proyectodw/src/main/java/com/example/proyectodw/model/Transaccion.java
package com.example.proyectodw.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToOne;

@Entity
public class Transaccion{
    @Id
    @GeneratedValue
    long ID;
    int tid;
    String tipo;
    int precioTotal;
    @ManyToOne
    Usuario usuario;

    public Transaccion() {
    }

    public Transaccion(int tid, String tipo, int precioTotal, Usuario usuario) {
        this.tid = tid;
        this.tipo = tipo;
        this.precioTotal = precioTotal;
        this.usuario = usuario;
    }

    

    
    public long getID() {
        return ID;
    }


    public int getPrecioTotal() {
        return precioTotal;
    }

    public void setPrecioTotal(int precioTotal) {
        this.precioTotal = precioTotal;
    }

    public Usuario getUsuario() {
        return usuario;
    }

    public void setUsuario(Usuario usuario) {
        this.usuario = usuario;
    }

    public int getTid() {
        return tid;
    }

    public void setTid(int tid) {
        this.tid = tid;
    }

    
    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }


}

File: src/app/model/transaccion.ts
import { Estrella } from './estrella';
import { Nave } from './nave';
import { Usuario } from './usuario';

export class Transaccion {
    public id: number = 0;
    public precioTotal: number = 0;
    public tid: number = 0;
    public tipo: string = "";
    public usuario: Usuario = new Usuario(0, "", "", "", "", 0, 0, 0, 
    new Nave(
        "", 0, 0, 0, 0, 0, new Estrella("", 0, 0, 0, 0, 0)
        )
    );


    constructor(id: number, precioTotal: number, tid: number, tipo: string, usuario: Usuario){
        this.id = id;
        this.precioTotal = precioTotal;
        this.tid = tid;
        this.tipo = tipo;
        this.usuario = usuario;
    }
}
